src/Core/Tool/Authorizer/UserSettingAuthorizer.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Ushahidi User Setting Authorizer
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\Core\Tool\Authorizer;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\User;
+use Ushahidi\Core\Entity\Permission;
+use Ushahidi\Core\Tool\Authorizer;
+use Ushahidi\Core\Traits\AdminAccess;
+use Ushahidi\Core\Traits\UserContext;
+use Ushahidi\Core\Traits\PrivAccess;
+use Ushahidi\Core\Traits\OwnerAccess;
+use Ushahidi\Core\Traits\PrivateDeployment;
+use Ushahidi\Core\Tool\Permissions\AclTrait;
+
+// The `UserAuthorizer` class is responsible for access checks on `Users`
+class UserSettingAuthorizer implements Authorizer
+{
+	// The access checks are run under the context of a specific user
+	use UserContext;
+
+	// - `AdminAccess` to check if the user has admin access
+	use AdminAccess, OwnerAccess;
+
+	// It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+	use PrivAccess;
+
+	// It uses `PrivateDeployment` to check whether a deployment is private
+	use PrivateDeployment;
+
+	// Check that the user has the necessary permissions
+	use AclTrait;
+
+	/**
+	 * Get a list of all possible privilges.
+	 * By default, returns standard HTTP REST methods.
+	 * @return Array
+	 */
+	protected function getAllPrivs()
+	{
+		return ['read', 'create', 'update', 'delete', 'search', 'read_full', 'register'];
+	}
+
+	/* Authorizer */
+	public function isAllowed(Entity $entity, $privilege)
+	{
+		// These checks are run within the user context.
+		$user = $this->getUser();
+
+		// Only logged in users have access if the deployment is private
+		if (!$this->canAccessDeployment($user)) {
+			return false;
+		}
+
+		// Role with the Manage Users permission can manage all users
+		if ($this->acl->hasPermission($user, Permission::MANAGE_USERS)) {
```

Comment:
I think maybe this shouldn't be true. ie. a user should be the only person who can edit their own settings? Admins don't need to be able to edit a users settings

Reply:
Not sure about that. Ideally, you should be able to see all user settings but I'm not sure really what benefit it is. I'll remove it.